src/component.cpp

```diff
@@ -142,12 +142,12 @@ std::string Component::doSerialisation(FORMATS format) const
         std::string componentName = getName();
         if (componentName.length()) {
             repr += " name=\"" + componentName + "\"";
-            if (variableCount() > 0) {
-                endTag = true;
-                repr += ">";
-                for(size_t i = 0; i < variableCount(); i++) {
-                    repr += getVariable(i)->serialise(format);
-                }
+        }
+        if (variableCount() > 0) {
+            endTag = true;
+            repr += ">";
+            for(size_t i = 0; i < variableCount(); i++) {
```

Comment:
I definitely prefer

`for (int i = 0; i < result->size(); i++) {`

much easier to read for me :) and I don't think we are really concerned with the performance hit in evaluating the "size" on each iteration in this code. (Similarly, I'm not worried about the performance hit in the pre- vs post-fix increment for simple integer data types, but can see that it would be best to be consistent and always go with the prefix increment operator)